File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    for (let i = 1; i<= yellow; i++) { // i yellow 세로
        if (!Number.isInteger(yellow / i)) { // 정수로 나누어지지않으면 continue
            continue;
        }
        const j = yellow / i // yellow 가로
        const row = j + 2 // 카펫 가로
        const col = i + 2 // 카펫 세로
        if (brown === row * 2 + i * 2) {
            return [row, col];
        }
    }
}

// [카펫 가로 row, 카펫 세로 col]
// 가로 >= 세로
// 노란색 : 중앙
// 갈색 : 테두리

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    for ysero in range(1, brown + yellow + 1):
        if yellow % ysero == 0:
            ygaro = yellow // ysero
            sero = ysero + 2
            garo = ygaro + 2

            if brown == (garo * 2 + ysero * 2):
                return [garo, sero]